src/flb_utils.c

```diff
@@ -1399,3 +1399,12 @@ int flb_utils_get_machine_id(char **out_id, size_t *out_size)
 
     return -1;
 }
+
+void flb_utils_str_property_check_dup(const char* key_name, char *prop, const char *new_prop)
```

Comment:
Please use the appropriate data types for `prop` and `new_prop` and replace `char*` with `char *` in `key_name` so it's consistent.

Please rename `prop` and `new_prop` so the names reflect that those are the old value and new value.
Please rename the function so its name clearly conveys what it does.

With all this in consideration this might be a better approach : 

```
void flb_utils_set_plugin_string_property(const char *name, 
                                          flb_sds_t *field_storage, 
                                          flb_sds_t  new_value)
{
    if (field_storage == NULL) {
        flb_error("[utils] invalid field storage pointer for property '%s'",
                  name);

        return;
    }

    if (*field_storage != NULL) {
        flb_warn("[utils] property '%s' is already specified with value '%s'."
                 " Overwriting with '%s'",
                 name, 
                 *field_storage, 
                 new_value);

        flb_sds_destroy(*field_storage);

        field_storage = NULL;
    }

    *field_storage = new_value;
}
```

Which means the client code would do something like this : 

```
flb_utils_set_plugin_string_property("tls.key_passwd", &ins->tls_key_passwd, tmp);
```

Which is more concise and clearly states its intention.

Reply:
I like this approach. Will fix with the suggestion

Reply:
Ah, your suggestion should be changed for re-initializing with NULL:
`field_storage = NULL;` should be `*field_storage = NULL;`.